packages/ui-plugin-docs/src/controllers/clipboard.controller.ts

```diff
@@ -1,20 +1,221 @@
+import {
+    Disposable,
+    ICommandInfo,
+    ICommandService,
+    IDocumentBody,
+    ILogService,
+    IParagraph,
+    ITextRun,
+    IUniverInstanceService,
+    Tools,
+} from '@univerjs/core';
+import { Inject } from '@wendellhu/redi';
 
-import { DocCopyCommand } from '../commands/commands/clipboard.command';
+import {
+    DocCopyCommand,
+    DocCutCommand,
+    DocPasteCommand,
+    InnerPasteCommand,
+} from '../commands/commands/clipboard.command';
 import { IDocClipboardService } from '../services/clipboard/clipboard.service';
 
 export class DocClipboardController extends Disposable {
     constructor(
+        @ILogService private readonly _logService: ILogService,
         @ICommandService private readonly _commandService: ICommandService,
         @IUniverInstanceService private readonly _currentUniverService: IUniverInstanceService,
-        @IDocClipboardService private readonly _docClipboardService: IDocClipboardService
+        @IDocClipboardService private readonly _docClipboardService: IDocClipboardService,
+        @ITextSelectionRenderManager private _textSelectionRenderManager: ITextSelectionRenderManager,
+        @Inject(TextSelectionManagerService) private _textSelectionManagerService: TextSelectionManagerService,
+        @Inject(DocSkeletonManagerService) private _docSkeletonManagerService: DocSkeletonManagerService
     ) {
         super();
+        this.commandExecutedListener();
     }
 
     initialize() {
-        [DocCopyCommand, DocCopyCommand, DocCopyCommand].forEach((command) =>
+        [DocCopyCommand, DocCutCommand, DocPasteCommand].forEach((command) =>
             this.disposeWithMe(this._commandService.registerAsMultipleCommand(command))
         );
+        [InnerPasteCommand].forEach((command) => this.disposeWithMe(this._commandService.registerCommand(command)));
+    }
+
+    private commandExecutedListener() {
+        const updateCommandList = [DocCutCommand.id, DocCopyCommand.id, DocPasteCommand.id];
+
+        this.disposeWithMe(
+            this._commandService.onCommandExecuted((command: ICommandInfo) => {
+                if (!updateCommandList.includes(command.id)) {
+                    return;
+                }
+
+                switch (command.id) {
+                    case DocPasteCommand.id: {
+                        this.handlePaste();
+                        break;
+                    }
+
+                    case DocCopyCommand.id: {
+                        this.handleCopy();
+                        break;
+                    }
+
+                    case DocCutCommand.id: {
+                        this.handleCut();
+                        break;
+                    }
+
+                    default:
+                        throw new Error(`Unhandled command ${command.id}`);
+                }
+            })
+        );
+    }
+
+    private async handlePaste() {
+        const { _docClipboardService: clipboard } = this;
```

Comment:
IMHO, these should be moved to DocPasteCommand.

Reply:
Why need moved this to DocPasteCommand？